ReactiveXComponent/XComponentApi.cs

```diff
@@ -9,19 +11,25 @@ public class XComponentApi : IXComponentApi
         private readonly IXCConnection _xcConnection;
         private readonly XCConfiguration _xcConfiguration;
 
-        private XComponentApi(Stream xcApiStream, string privateCommunicationIdentifier = null)
+        public static string PrivateCommunicationIdentifier;
+
+        private XComponentApi(Stream xcApiStream)
         {
-            var parser = new Parser.Parser(xcApiStream);
+            var parser = new XCApiConfigParser();
             _xcConfiguration = new XCConfiguration(parser);
-            _xcConfiguration.Init();
-            var connectionFactory = new XCConnectionFactory(_xcConfiguration);
+            _xcConfiguration.Init(xcApiStream);
+            AbstractXCConnectionFactory connectionFactory = new XCConnectionFactory(_xcConfiguration);
             _xcConnection = connectionFactory.CreateConnection();
-            XCPublisher.PrivateCommunicationIdentifier = privateCommunicationIdentifier;
         }
 
-        public static IXComponentApi CreateFromXCApi(Stream xcApiStream, string privateCommunicationIdentifier = null)
+        public static void InitPrivateCommunivation(string privateCommunicationIdentifier)
```

Comment:
add privateCommunication as an option parameter of the static factory method CreateFromXCApi